src/main/java/com/example/W5Assessment/service/TaskServiceImpl.java

```diff
@@ -0,0 +1,82 @@
+package com.example.W5Assessment.service;
+
+import com.example.W5Assessment.dao.TaskRepository;
+import com.example.W5Assessment.entity.Task;
+import com.example.W5Assessment.exception.EmptyTaskListException;
+import com.example.W5Assessment.exception.NoSuchTaskExistsException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Service
+public class TaskServiceImpl implements TaskService{
+    private final TaskRepository taskRepository;
+
+    public TaskServiceImpl(TaskRepository taskRepository) {
+        this.taskRepository = taskRepository;
+    }
+
+    @Override
+    public List<Task> findAll() {
+        if (taskRepository.findAllTaskExist() != null) {
```

Comment:
there is no need for this check, just return findAll() , if its empty its empty. You are calling the same query two times, makes the performance bad